source/azure_iot_hub_client.c

```diff
@@ -462,6 +503,56 @@ static AzureIoTHubClientResult_t prvWaitForSubAck( AzureIoTHubClient_t * pxAzure
 
     return xResult;
 }
+
+/**
+ * Do blocking wait for pub-ack of particular publish.
+ *
+ **/
+static AzureIoTHubClientResult_t prvWaitForPubAck( AzureIoTHubClient_t * pxAzureIoTHubClient,
+                                                   AzureIoTHubClientTelemetryAckContext_t * pxContext,
+                                                   uint32_t ulTimeoutMilliseconds )
+{
+    AzureIoTHubClientResult_t xResult = eAzureIoTHubClientSubackWaitTimeout;
+    uint32_t ulWaitTime;
+
+    AZLogDebug( ( "Waiting for puback for packet id: 0x%08x", pxContext->_internal.usMqttPubPacketID ) );
+
+    do
+    {
+        if( pxContext->_internal.usState == azureiothubTELEMETRY_ACK_STATE_PUBACK )
+        {
+            xResult = eAzureIoTHubClientSuccess;
+            break;
+        }
+
+        if( ulTimeoutMilliseconds > azureiotPUBACK_WAIT_INTERVAL_MS )
+        {
+            ulTimeoutMilliseconds -= azureiotPUBACK_WAIT_INTERVAL_MS;
+            ulWaitTime = azureiotPUBACK_WAIT_INTERVAL_MS;
+        }
+        else
+        {
+            ulWaitTime = ulTimeoutMilliseconds;
+            ulTimeoutMilliseconds = 0;
+        }
+
+        if( AzureIoTMQTT_ProcessLoop( &pxAzureIoTHubClient->_internal.xMQTTContext, ulWaitTime ) != eAzureIoTMQTTSuccess )
+        {
+            xResult = eAzureIoTHubClientFailed;
+            break;
+        }
+    } while( ulTimeoutMilliseconds );
```

Comment:
is it possible to merge this code with above code?

Reply:
please take a look at the most recent changes. I've merged the `waitForX` stuff into one and it means consolidation of other things like here: https://github.com/Azure/azure-iot-middleware-freertos/blob/7778f751e10309f4ff6e89f499d27f677ff61521/source/azure_iot_hub_client.c#L36-L41

and here:
https://github.com/Azure/azure-iot-middleware-freertos/blob/7778f751e10309f4ff6e89f499d27f677ff61521/source/include/azure_iot_hub_client.h#L54